### DISCUSSION TITLE
Number Input Custom Format and Parsing - the suffix/prefix is included in parsing

### DISCUSSION THREAD

**Discussion by domi-glyph** [chakra-ui/chakra-ui#8801] [General]
### Description

I wanted to have suffix the word 'pieces'. When typing in number into the field. The field will generate "ee".

### Link to Reproduction

https://v2.chakra-ui.com/docs/components/number-input/usage#formatting-and-parsing-the-value

### Steps to reproduce

1. Go to Section "Formatting and Parsing the value" [https://v2.chakra-ui.com/docs/components/number-input/usage#formatting-and-parsing-the-value](url)
2. Input this code
```html
function Example() {
  const format = (val) => val + ' pieces'
  const parse = (val) => val.replace(/ pieces$/, '')

  const [value, setValue] = React.useState('1')

  return (
    <NumberInput
      onChange={(valueString) => setValue(parse(valueString))}
      value={format(value)}
    >
      <NumberInputField />
      <NumberInputStepper>
        <NumberIncrementStepper />
        <NumberDecrementStepper />
      </NumberInputStepper>
    </NumberInput>
  )
}
``` 
4. Try input number
5. See error

### Chakra UI Version

2.8.2

### Operating System

- [x] macOS
- [x] Windows
- [x] Linux

### Additional Information

I suspect as there is 'e' to use as exponent of base 10 in number. It parses the word 'pieces' that contains 2 e. Hence it appears 'ee' in every typing.

**Comment by domi-glyph** [comment]
For now i'm solving it as below. I don't think it is a real solution, as it will prevent 'e' or any other letters.
```html
function Example() {
  const format = (val) => val + ' pieces'
  const parse = (val) => val.replace(/[a-zA-Z]+/g, '')

  const [value, setValue] = React.useState('1.53')

  return (
    <NumberInput
      onChange={(valueString) => {
        setValue(parse(valueString))
      }}
      value={format(value)}
    >
      <NumberInputField />
      <NumberInputStepper>
        <NumberIncrementStepper />
        <NumberDecrementStepper />
      </NumberInputStepper>
    </NumberInput>
  )
}
```

**Comment by phoneticallySAARTHaK** [comment]
You can use an InputGroup for that

**Comment by segunadebayo** [comment]
v3 NumberInput does not have `format` and `parse`. Put the suffix in `InputAddon` and parse the raw number yourself.
